### DISCUSSION TITLE
Unable to start Netbox when Plugin configured

### DISCUSSION THREAD

**Discussion by benek984** [netbox-community/netbox#12261] [Help Wanted!] [answered]
I installed plugin netbox_devicetype_importer added it to the config file

````
PLUGINS = ['netbox_devicetype_importer']

PLUGINS_CONFIG = {
    'netbox_devicetype_importer': {
        'github_token': 'gxxxxxHxxxxxxxxxxezefdJ9pggxxxxxxxxs8'
    }
 }

````

When restarting Netbox it doesn't start with error:

````
[root@nbox01 netbox]# systemctl status netbox
● netbox.service - NetBox WSGI Service
   Loaded: loaded (/etc/systemd/system/netbox.service; enabled; vendor preset: disabled)
   Active: activating (auto-restart) (Result: exit-code) since Fri 2023-04-14 09:00:06 PDT; 8s ago
     Docs: https://docs.netbox.dev/
  Process: 169755 ExecStart=/opt/netbox/venv/bin/gunicorn --pid /var/tmp/netbox.pid --pythonpath /opt/netbox/netbox --config /opt/netbox/gunicorn.py netbox.wsgi (code=exited, status=1/FAILURE)
 Main PID: 169755 (code=exited, status=1/FAILURE)

Apr 14 09:00:06 nbox01 systemd[1]: netbox.service: Failed with result 'exit-code'.

````
Can you guys help?

**Comment by candlerb** [comment]
First step would be looking at the error message: `journalctl -eu netbox`

But my guess would be you didn't activate the venv before installing the plugin package.

Another approach is to create a file `local_requirements.txt` containing the name of the package (i.e. just `netbox-devicetype-importer`) and then run `upgrade.sh` as if you were updating Netbox. This has the advantage that the package will be reinstalled automatically every time you upgrade.

**Reply by benek984** [reply]
I definetely did not activate venv as I dont know how? can you advise how to ?
Regardsing this local_requirments.txt shoudl I just add one line like this to it?

````
netbox-devicetype-importer
````

**Reply by candlerb** [reply]
You haven't shown the output of `journalctl -eu netbox`. This should make it clear whether this is the problem or not.

Activating the virtualenv to install a package:

* `sudo -s` to get a root shell (if you're not already root)
* `source /opt/netbox/venv/bin/activate`, as mentioned [here](https://docs.netbox.dev/en/stable/installation/3-netbox/#create-a-super-user)
* `pip install ....` as required
* `deactivate` to disable the venv
* `exit` to exit the root shell

Yes, the contents of `local_requirements.txt` should be just the name of the package, as you'd use with `pip install`.

**Reply by benek984** [reply]
Performed all the steps you sugested but still the same please see below for the output of ``` journalctl -eu netbox```
Last few lines
````
Apr 17 02:44:05 ussd-prd-nbox01.illumina.com gunicorn[193978]:     raise ImproperlyConfigured(
Apr 17 02:44:05 ussd-prd-nbox01.illumina.com gunicorn[193978]: django.core.exceptions.ImproperlyConfigured: Plugin netbox_devicetype_importer requires NetBox maximum version 3.3.99.
Apr 17 02:44:05 ussd-prd-nbox01.illumina.com gunicorn[193978]: [2023-04-17 02:44:05 -0700] [193978] [INFO] Worker exiting (pid: 193978)
Apr 17 02:44:05 ussd-prd-nbox01.illumina.com systemd[1]: netbox.service: Failed with result 'exit-code'.
`````

**Reply by kkthxbye-code** [reply, marked as answer]
It says exactly what is wrong in what you pasted:

> Plugin netbox_devicetype_importer requires NetBox maximum version 3.3.99

https://github.com/k01ek/netbox-devicetype-importer/issues/21

**Reply by abhi1693** [reply]
You can try this https://github.com/Onemind-Services-LLC/netbox-metatype-importer. We keep this pretty much up to date and is a maintained fork of the original plugin

**Reply by benek984** [reply]
@abhi1693 thanks, I did installed this after reading the link from @kkthxbye-code. Thanks all for help.
